lib/sources.nix

```diff
@@ -26,14 +26,29 @@ rec {
     (type == "symlink" && lib.hasPrefix "result" baseName)
   );
 
-  cleanSource = builtins.filterSource cleanSourceFilter;
+  cleanSource = src: filterSource { filter = cleanSourceFilter; inherit src; };
+
+  # Like `builtins.filterSource`, except it will compose with itself,
+  # allowing you to chain multiple calls together without any
+  # intermediate copies being put in the nix store.
+  filterSource = { filter, src }:
```

Comment:
Please don't use the same name as a builtin function. Call it `composableFilterSource` or something like that.

Reply:
@peti @edolstra How about `cleanSourceWith`? Avoids the naming clash but remains generic.

Reply:
`cleanSourceWith` sounds good to me.

Reply:
Done